src/theory/quantifiers/sygus/synth_conjecture.cpp

```diff
@@ -1059,6 +1060,18 @@ bool SynthConjecture::getSynthSolutionsInternal(std::vector<Node>& sols,
     }
     d_sol.push_back(sol);
     d_solStatus.push_back(status);
+    // Note that this assumes that the name of the resulting datatype matches
```

Comment:
Yes, it will just lead to confusing names.

When we reconstruct from single invocation, the annotation should work correctly, since reconstruction will generate the sygus datatype term.

I'll clarify.